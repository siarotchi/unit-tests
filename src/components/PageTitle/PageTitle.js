import React from 'react';
import PropTypes from 'prop-types';

import './PageTitle.scss'

const PageTitle = ({ label }) => (
  <h1 className="page-title" data-testid="page-title">
    {label}
  </h1>
);

PageTitle.propTypes = {
  label: PropTypes.node.isRequired,
};

export default PageTitle;
