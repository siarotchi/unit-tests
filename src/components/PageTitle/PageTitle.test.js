import React from "react";
import PageTitle from "./PageTitle";
import { render } from "@testing-library/react";

describe("<PageTitle />", function () {
  it("should match the snapshot", () => {
    /**
     * Write a snapshoâ—˜t test for PageTitle component
     * Don't forget to add snapshot file to the git
     */

    const labelText = "label text";
    const { container, getByText } = render(<PageTitle label={labelText} />);

    expect(getByText(labelText)).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});
