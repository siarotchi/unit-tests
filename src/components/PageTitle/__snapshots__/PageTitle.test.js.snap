// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PageTitle /> should match the snapshot 1`] = `
<div>
  <h1
    class="page-title"
    data-testid="page-title"
  >
    label text
  </h1>
</div>
`;
